src/Microsoft.SymbolManifestGenerator/SymbolManifestGenerator.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SymbolStore;
+using Microsoft.SymbolStore.KeyGenerators;
+
+namespace Microsoft.SymbolManifestGenerator
+{
+    public class SymbolManifestGenerator
+    {
+        private const int Private = 340;
+        public SymbolManifestGenerator()
+        {
+        }
+
+        public static void GenerateManifest(ITracer tracer, DirectoryInfo dir, string manifestFileName)
+        {
+            try
+            {
+                ManifestDataV1 manifestData = new();
+
+                // Iterate through all files recursively in the directory
+                foreach (FileInfo file in dir.GetFiles("*", SearchOption.AllDirectories))
+                {
+                    string relativePath = file.DirectoryName.Replace(dir.FullName, string.Empty).TrimStart(Path.DirectorySeparatorChar);
+                    FileStream fileStream = file.OpenRead();
+                    SymbolStoreFile symbolStoreFile = new(fileStream, file.FullName);
+                    FileKeyGenerator generator = new(tracer, symbolStoreFile);
+                    IEnumerable<SymbolStoreKey> clrKeys = generator.GetKeys(KeyTypeFlags.ClrKeys);
+                    foreach (SymbolStoreKey clrKey in clrKeys)
+                    {
+                        string additionalDebugEntryFileName = clrKey.FullPathName;
+                        string basedirRelativePath = Path.Combine(relativePath, additionalDebugEntryFileName);
+                        string additionalDebugEntryFilePath = Path.Combine(dir.FullName, basedirRelativePath);
+                        FileInfo additionalDebugEntryFileInfo = new(additionalDebugEntryFilePath);
+                        if (!File.Exists(additionalDebugEntryFilePath))
+                        {
+                            tracer.Warning($"File {basedirRelativePath} does not exist. Skipping.");
+                        }
+                        string fileHash = CalculateSHA512(additionalDebugEntryFileInfo);
+
+                        ManifestDataEntry manifestDataEntry = new()
+                        {
+                            basedirRelativePath = basedirRelativePath,
+                            symbolKey = clrKey.Index,
+                            sha512 = fileHash,
+                            debugInformationLevel = Private
+                        };
+                        manifestData.entries.Add(manifestDataEntry);
+                    }
+                }
+
+                string manifestDataContent = JsonSerializer.Serialize(manifestData);
+                File.WriteAllText(manifestFileName, manifestDataContent);
+            }
+            catch (Exception ex)
```

Comment:
How does the calling code know that the manifest generation failed?

Using Console.WriteLine below should be tracer.Error or something other than Console.

Reply:
Ah I added the try catch because I was hitting some exception for adding the wrong version for System.Text.Json, it was for debugging, wasn't intention to keep the try catch.